cheetah/accelerator/space_charge_kick.py

```diff
@@ -506,9 +505,9 @@ def _compute_forces(
         # Keep dimensions, and set F to zero if non-valid
         force_indices = (
             idx_vector,
-            torch.clamp(idx_x, max=grid_shape[0] - 1),
-            torch.clamp(idx_y, max=grid_shape[1] - 1),
-            torch.clamp(idx_tau, max=grid_shape[2] - 1),
+            torch.clamp(idx_x, min=0, max=grid_shape[0] - 1),
+            torch.clamp(idx_y, min=0, max=grid_shape[1] - 1),
+            torch.clamp(idx_tau, min=0, max=grid_shape[2] - 1),
         )
```

Comment:
@RemiLehe @cr-xu I got negative indices here, which caused issues with a beam from ARES that I imported from Astra. Is the clamping "legal" or am I masking a different issue by doing that.

In addition, @cr-xu, you added that comment. I'm not sure I fully understand it. Is it correct / does it match what the code does?

Reply:
I don't immediately see the issue here, if the test passes then it should be fine... 
Anyway I'm surprised that negative indices even appear. 

The comment refers to L507-518, basically masking the forces so that they come from only real particles. I 'm not sure if we need to move it.